src/compiler/checker.ts

```diff
@@ -6609,7 +6611,7 @@ module ts {
             }
 
             // The identityMapper object is used to indicate that function expressions are wildcards
-            if (contextualMapper === identityMapper) {
+            if (contextualMapper === identityMapper && isContextSensitive(node)) {

```

Comment:
Previously we would never pass identityMapper during inferential typing and it was fine to just treat *all* function expressions in the argument as wildcards. We now want a "hybrid" type constructed where only context sensitive function expessions are blanked out, such that we can infer from it in the first pass.